src/schema/schema_open.c

```diff
@@ -297,8 +297,19 @@ __schema_open_index(WT_SESSION_IMPL *session,
 	for (i = 0; ret == 0; i++, ret = cursor->next(cursor)) {
 		WT_ERR(cursor->get_key(cursor, &uri));
 		name = uri;
-		if (!WT_PREFIX_SKIP(name, tmp->data))
+
+		if (!WT_PREFIX_SKIP(name, tmp->data)) {
+			/*
+			 * We reached the end of index list, remove the rest of
+			 * in memory indices, they no longer exist.
+			 */
+			while (i < table->nindices) {
```

Comment:
I am not sure if I should only do this if "idxname == NULL" which would imply that we are scanning all the indices.

Reply:
It makes sense to me to do the removal in all cases, not just when `idxname == NULL`.